modules/packages/ParallelIO.chpl

```diff
@@ -0,0 +1,705 @@
+            locBounds = byteOffsets[id]..byteOffsets[id+1],
+            locFile = open(filePath, ioMode.r),
+            tByteOffsets = findDelimChunks(locFile, delim, tasksPerLoc, locBounds, headerPolicy.noHeader),
+            tItemOffsets = findItemOffsets(locFile, delim, tByteOffsets) + itemOffsets[id];
+
+      coforall tid in 0..<tasksPerLoc with (ref results) {
+        var des: deserializerType;
+        const taskBounds = tByteOffsets[tid]..tByteOffsets[tid+1],
+              r = locFile.reader(locking=false, region=taskBounds, deserializer=des);
+
+        var item: t, first = true;
+        for i in tItemOffsets[tid]..<tItemOffsets[tid+1] {
+          if first then first = false; else r.advanceThrough(delim);
+          r.read(item);
+          results[i] = item;
+        }
+      }
+    }
+
+    return results;
+  }
+
+  /*
+    Read a delimited file in parallel into an array.
+
+    This procedure is essentially the same as :proc:`readParallelDelimited`,
+    except that it only executes on the calling locale. As such, it does not
+    accept a ``targetLocales`` argument and returns a non-distributed array.
+
+    :arg filePath: a path to the file to read from
+    :arg delim: the delimiter to use to separate ``t`` values in the file
+    :arg t: the type of value to read from the file
+    :arg nTasks: the number of tasks to use
+    :arg header: how to handle the file header (see :record:`headerPolicy`)
+    :arg deserializerType: the type of deserializer to use
+
+    :returns: a default rectangular array of ``t`` values
+
+    :throws: :class:`OffsetNotFoundError` if a starting offset cannot be found in
+              any of the chunks
+
+    See :proc:`~IO.open` for other errors that could be thrown when attempting
+    to open the file
+  */
+  proc readParallelDelimitedLocal(filePath: string, type t, in delim: ?dt = b"\n",
+                                  header = headerPolicy.noHeader, nTasks: int = here.maxTaskPar,
+                                  type deserializerType = defaultDeserializer
+  ): [] t throws
+    where dt == bytes || dt == string
+  {
+    const f = open(filePath, ioMode.r),
+          fileBounds = 0..<f.size,
+          byteOffsets = findDelimChunks(f, delim, nTasks, fileBounds, header),
+          itemOffsets = findItemOffsets(f, delim, byteOffsets);
+
+    const resDom = {0..<itemOffsets.last};
+    var results: [resDom] t;
+
+    coforall tid in 0..<nTasks with (ref results) {
+      var des: deserializerType;
+      const taskBounds = byteOffsets[tid]..<byteOffsets[tid+1],
+            r = f.reader(locking=false, region=taskBounds, deserializer=des);
+
+      var item: t, first = true;
+      for i in itemOffsets[tid]..<itemOffsets[tid+1] {
+        if first then first = false; else r.advanceThrough(delim);
+        r.read(item);
+        results[i] = item;
+      }
+    }
+
+    return results;
+  }
+
+  /*
+    Read a delimited file in parallel into a block-distributed array.
```

Comment:
The docs here don't explain why this is different from readParallelDelimited, but your PR message does.

Reply:
The difference comes in the next paragraph. I'll add something to the first sentence that makes them distinct (and probably also come up with better names for the procedures).